### PR TITLE
[Merged by Bors] - tortoise: reload state in background when node is started

### PR DESCRIPTION
**Pull request by dshulyak**
## Motivation

Removing persistent state from tortoise greatly simplifies code and further changes. It is also fast enough to recover it from original databases (mesh and atx) when we start the node.

on cloud nodes, it takes 2minutes to load 10 000 layers. on a node with a faster hardware, specifically disk, it takes 50s. verifying tortoise, from mentioned pr, loads same amount of layers in 20s.

prereq for https://github.com/spacemeshos/go-spacemesh/pull/2991

## Changes
- load state in background when node is started
- wait for tortoise to become ready before starting sync

## Test Plan
unit and manual tests

## TODO
- [x] Explain motivation or link existing issue(s)
- [x] Test changes and document test plan
- [x] Update documentation as needed

## DevOps Notes

- [x] This PR does not require configuration changes (e.g., environment variables, GitHub secrets, VM resources)
- [x] This PR does not affect public APIs
- [x] This PR does not rely on a new version of external services (PoET, elasticsearch, etc.)
- [x] This PR does not make changes to log messages (which monitoring infrastructure may rely on)

### CODE REVIEW

File: tortoise/algorithm.go [hunk 1]
```diff
@@ -110,37 +113,43 @@ func New(db database.Database, mdb blockDataProvider, atxdb atxDataProvider, bea
...
+	if needsRecovery {
+		t.trtl.Processed = t.cfg.MeshProcessed
+		// TODO(dshulyak) last should be set according to the clock.
+		t.trtl.Last = t.cfg.MeshProcessed
+
+		t.logger.Info("loading state from disk. make sure to wait until tortoise is ready",
+			log.Stringer("last_layer", t.cfg.MeshProcessed),
+		)
+		t.eg.Go(func() error {
+			t.ready <- t.rerun(ctx)
```
Comment (countvonzero) [inline tortoise/algorithm.go:139]: can you document in the PR how long does this take with mesh growth? thanks

File: mesh/mesh.go [hunk 2]
```diff
@@ -159,21 +157,10 @@ func NewRecoveredMesh(ctx context.Context, db *DB, atxDb AtxDB, rewardConfig Con
 	if err != nil {
 		logger.With().Panic("failed to load state for layer", msh.LatestLayerInState(), log.Err(err))
 	}
-	// in case we load a state that was not fully played
-	nextLayer := msh.LatestLayerInState().Add(1)
-	lastComplete := msh.trtl.LatestComplete()
-	if nextLayer.Before(msh.trtl.LatestComplete()) {
```
Comment (countvonzero) [inline mesh/mesh.go:165]: say for whatever reason, the rerun on startup yields a different verified layer before the restart. and the verified layer is different from the layer applied layer in state. i think we need to double check that too.

**Review by countvonzero — APPROVED** [review]
so much nicer.

File: mesh/mesh.go (hunk 2 above)
Comment (dshulyak) [reply mesh/mesh.go:165]: that would be a standard revert flow. this can be expected because we may load more data when node is re-started

**Comment by dshulyak** [comment]
bors merge

File: mesh/mesh.go (hunk 2 above)
Comment (countvonzero) [reply mesh/mesh.go:165]: but how about the verified layer is higher than before the restart? say before the restart the state is applied up to N. but after restart, rerun verified up to N+10. how will the state catch up and apply TXs in the extra 10 layers? when the next hare call `HandleIncomingLayer`?

Comment (dshulyak) [reply mesh/mesh.go:165]: for example processed = 20, verified = 10. both processed and verified are loaded from meshdb, so all the data is already there. when recover starts it turns out that we can verify up to 19. so we return old = 10, new = 19 to mesh. and it applies everything in between.

sync will call HandleIncomingLayer from 21 and onward. Hare is more likely from current and onward, and current maybe 21 but unlikely.

**Pull request closed without merging**